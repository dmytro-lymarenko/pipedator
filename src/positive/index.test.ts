import { positive } from './index';

describe('positive()', () => {
	it('should validate', () => {
		expect(positive().validate(4)).toBeNull();
	});

	it('should fail', () => {
		expect(positive().validate(0)).toMatchSnapshot();
		expect(positive().validate(-4)).toMatchSnapshot();
	});

	it('should support custom message', () => {
		expect(positive('Custom message').validate(-1)).toMatchSnapshot();
	});
});
