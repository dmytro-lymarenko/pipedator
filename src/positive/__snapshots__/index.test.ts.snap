// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`positive() should fail 1`] = `
Object {
  "children": null,
  "message": "Value should be positive",
  "path": "",
}
`;

exports[`positive() should fail 2`] = `
Object {
  "children": null,
  "message": "Value should be positive",
  "path": "",
}
`;

exports[`positive() should support custom message 1`] = `
Object {
  "children": null,
  "message": "Custom message",
  "path": "",
}
`;
